src/core/net/netif.cpp

```diff
@@ -430,7 +430,7 @@ Error Netif::UnsubscribeExternalMulticast(const Address &aAddress)
     entry = mMulticastAddresses.FindMatching(aAddress, prev);
     VerifyOrExit(entry != nullptr, error = kErrorNotFound);
 
-    VerifyOrExit(IsMulticastAddressExternal(*entry), error = kErrorInvalidArgs);
+    VerifyOrExit(IsMulticastAddressExternal(*entry), error = kErrorRejected);
```

Comment:
nit: add this new retval in the header files?

and should we add check if it's a multicast address if we want to keep the retval `kErrorInvalidArgs`?

Reply:
done